src/main/java/graphql/language/PrettyPrinter.java

```diff
@@ -0,0 +1,462 @@
+package graphql.language;
+
+import graphql.PublicApi;
+import graphql.collect.ImmutableKit;
+import graphql.parser.CommentParser;
+import graphql.parser.NodeToRuleCapturingParser;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static graphql.Assert.assertTrue;
+
+/**
+ * A printer that acts as a code formatter.
+ *
+ * This printer will preserve pretty much all elements from the source text, even those that are not part of the AST
+ * (and are thus discarded by the {@link AstPrinter}), like comments.
+ *
+ * @see AstPrinter
+ */
+@PublicApi
+public class PrettyPrinter extends AstPrinter {
```

Comment:
What about PrettyAstPrinter as a name

We have today an `AstPrinter` and a `SchemaPrinter` - so this printer is a `PrettyAstPrinter` or `AstPrettyPrinter`

Reply:
100%!
Just pushed a commit with the change.